aea/protocols/generator.py

```diff
@@ -76,6 +82,79 @@ def _extract_all_contents(self) -> Dict[str, Dict[str, str]]:
                 all_contents[performative][content_name] = content_type
         return all_contents
 
+    def _specification_type_to_python_type(self, specification_type: str) -> str:
```

Comment:
This function would benefit from being broken up into shorter functions so it becomes more readable.

Reply:
For a start try splitting out each block in an `if, elif, else` and see what that gives you.

Reply:
The code is recursive so hard to split. We probably end up with more code since we need another function that manages calls to smaller functions etc.